src/main/kotlin/com/coxautodev/graphql/tools/SchemaParser.kt

```diff
@@ -128,7 +128,8 @@ class SchemaParser internal constructor(scanResult: ScannedSchemaObjects, privat
 
         val additionalObjects = objects.filter { o -> o != query && o != subscription && o != mutation }
 
-        return SchemaObjects(query, mutation, subscription, (additionalObjects + inputObjects + enums + interfaces + unions).toSet(), codeRegistryBuilder)
+        @Suppress("UNCHECKED_CAST") val dictionary = (additionalObjects + inputObjects + enums + interfaces + unions).toSet() as Set<GraphQLType>
```

Comment:
We can get rid off the warning suppression in various ways. I would recommend something like this:

```suggestion
        val types = (objects as Set<GraphQLType>) + inputObjects + enums + interfaces + unions

        return SchemaObjects(query, mutation, subscription, types, codeRegistryBuilder)
```

Reply:
The suggested code produces the same warning - "Unchecked cast":

![Screenshot 2020-01-21 at 22 02 12](https://user-images.githubusercontent.com/459215/72842935-cfab6300-3c99-11ea-84b6-d7311a9cc230.png)

Reply:
Right, it needs to be:
```
val types = (objects.toSet() as Set<GraphQLType>) + inputObjects + enums + interfaces + unions
```

Reply:
Done. Thanks!